messaging/src/main/java/org/axonframework/messaging/deadletter/InMemoryDeadLetterQueue.java

```diff
@@ -0,0 +1,373 @@
+     * to {@code 1024}, the dead letter expire threshold defaults to a {@link Duration} of 5000 milliseconds, and the
+     * {@link ScheduledExecutorService} defaults to a {@link Executors#newSingleThreadScheduledExecutor(ThreadFactory)},
+     * using an {@link AxonThreadFactory}.
+     *
+     * @param <T> The type of {@link Message} maintained in this {@link DeadLetterQueue}.
+     */
+    public static class Builder<T extends Message<?>> extends SchedulingDeadLetterQueue.Builder<Builder<T>, T> {
+
+        private int maxQueues = 1024;
+        private int maxQueueSize = 1024;
+
+        /**
+         * Sets the maximum number of queues this {@link DeadLetterQueue} may contain. This requirement reflects itself
+         * as the maximum amount of unique {@link QueueIdentifier QueueIdentifiers} referencing a dead-letter queue.
+         * <p>
+         * The given {@code maxQueues} is required to be a positive number, higher or equal to {@code 128}. It defaults
+         * to {@code 1024}.
+         *
+         * @param maxQueues The maximum amount of queues for the queue under construction.
+         * @return The current Builder, for fluent interfacing.
+         */
+        public Builder<T> maxQueues(int maxQueues) {
+            assertThat(maxQueues,
+                       value -> value >= 128,
+                       "The maximum number of queues should be larger or equal to 128");
+            this.maxQueues = maxQueues;
+            return this;
+        }
+
+        /**
+         * Sets the maximum amount of {@link DeadLetter letters} per queue this {@link DeadLetterQueue} can store.
+         * <p>
+         * The given {@code maxQueueSize} is required to be a positive number, higher or equal to {@code 128}. It
+         * defaults to {@code 1024}.
+         *
+         * @param maxQueueSize The maximum amount of {@link DeadLetter letters} per queue.
+         * @return The current Builder, for fluent interfacing.
+         */
+        public Builder<T> maxQueueSize(int maxQueueSize) {
+            assertThat(maxQueueSize,
+                       value -> value >= 128,
+                       "The maximum number of letters in a queue should be larger or equal to 128");
+            this.maxQueueSize = maxQueueSize;
+            return this;
+        }
+
+        /**
+         * {@inheritDoc}
+         * <p>
+         * The provided threshold also defines when newly {@link #enqueue(QueueIdentifier, Message, Throwable) enqueued}
+         * letters are considered ready to be {@link #take(String) taken}.
+         *
+         * @param expireThreshold The threshold for scheduling
+         *                        {@link #onAvailable(String, Runnable) availability checks} and for enqueued
+         *                        {@link DeadLetter letters} to be considered ready to be {@link #take(String) taken}.
+         */
+        @Override
+        public Builder<T> expireThreshold(Duration expireThreshold) {
```

Comment:
would this be the "retryInterval", or enqueueDuration?
"Threshold" doesn't seem to be the right term for what we're doing here.

Reply:
Note taken. This should be called differently, but that'll be part of the API changes around retrying/evaluating dead letters I wager.

Reply:
This field is no longer present, nor configurable, as the entire retry/release/process solution has been reimagined. Hence I am resolving this comment.